### PR TITLE
Use CalMuonDetectorPhases in frequency domain analysis GUI

### LINKED ISSUE #21419: FDA use calMuonDetecotrPhases
The phaseQuad option in the FDA GUI should use CalMuonDetectorPhases to generate the phase table.

### PR DESCRIPTION
**Pull request by AnthonyLim23**
The frequency domain analysis GUI currently (v 3.11) uses a rough calculation for the phase table. This replaces the rough calculation with the algorithm CalMuonDetectorPhases. This allows for a more robust creation of the phase table. 

The phase table is used when combining the frequency spectra. 

**To test:**
1. Open Muon analysis and select MUSR as the instrument.
2. Enter 62260 as the run number (some data should then load) 
3. Expand the group workspace MUSR00062260
4. Close muon analysis
5. Open the frequency domain analysis GUI (in muon)
6. Change the workspace to phaseQuad
7. Change the first good data to 0.5
8. press the calculate FFT button. A phaseQuad workspace will appear in MUSR00062260
9. Open the phaseTable and change one of the phases (you will have to make it non-read only)
10. Untick the construct phase table option
11. Press the calculate FFT button
12. Check the phaseTable - your change should  still be there
13. Tick the construct phase table option
14. press the calculate FFT button 
15. Open the phaseTable and it will have reverted. 

Fixes #21419. 

**Release Notes** 

In the release notes
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/fft_model.py [hunk 1]
```diff
@@ -42,19 +41,22 @@ def execute(self):
...
-            self.model.preAlg(self.preIm)
+            if self.preIm is not None:
+                self.model.preAlg(self.preIm)
 
-        if self.FFT is not None:
-            self.model.FFTAlg(self.FFT)
+            if self.FFT is not None:
+                self.model.FFTAlg(self.FFT)
+        except:
+            pass
```
Comment (josephframsay) [inline scripts/Muon/fft_model.py]: Would it not be better to show some kind of warning that an error occurred, as opposed to just failing silently? Or does this happen elsewhere?

File: scripts/Muon/fft_model.py [hunk 2]
```diff
@@ -100,50 +102,28 @@ def FFTAlg(self,FFTInputs):
...
-                phi=math.atan2(det.Z(),det.Y())
-                asym=math.sqrt(det.Z()**2+det.Y()**2)/r
-            elif(axis=="y"):
-                phi=math.atan2(det.X(),det.Z())
-                asym=math.sqrt(det.X()**2+det.Z()**2)/r
-            else: # z
-                phi=math.atan2(det.Y(),det.X())
-                asym=math.sqrt(det.Y()**2+det.X()**2)/r
-            phaseTable.addRow([j,asym,phi])
+        alg=mantid.AlgorithmManager.create("CalMuonDetectorPhases")
```
Comment (josephframsay) [inline scripts/Muon/fft_model.py]: The lack of whitespace makes this code quite dense. [PEP8](https://www.python.org/dev/peps/pep-0008/#other-recommendations) recommends a space around the `=` operator for readability. Would it be possible to apply formatting to these changes to make them a bit more readable?

File: scripts/Muon/fft_view.py [hunk 3]
```diff
@@ -53,14 +53,18 @@ def __init__(self, parent=None):
         table_utils.setRowName(self.FFTTable,5,"Use Raw data")
         self.Raw_box= table_utils.addCheckBoxToTable(self.FFTTable,True,5)
 
-        options=['x','y','z']
-        table_utils.setRowName(self.FFTTable,6,"Axis")
-        self.axis= table_utils.addComboToTable(self.FFTTable,6,options)
+        table_utils.setRowName(self.FFTTable,6,"First Good Data")
```
Comment (josephframsay) [inline scripts/Muon/fft_view.py]: Similarly, a space after the commas in the argument list would make this easier to parse

File: scripts/test/Muon/FFTPresenter_test.py [hunk 4]
```diff
@@ -35,7 +35,8 @@ def setUp(self):
         self.view.getFFTRePhase=mock.Mock()
         self.view.getFFTImPhase=mock.Mock()
         self.view.getWS=mock.Mock(return_value="MUSR00023456")
-        self.view.getAxis=mock.Mock(return_value="x")
+        self.view.getFirstGoodData=mock.Mock(return_value=0.1)
```
Comment (josephframsay) [inline scripts/test/Muon/FFTPresenter_test.py]: The asserts and assignments in this file would also be much more readable with a few extra spaces

**Review by josephframsay — CHANGES_REQUESTED** [review]
These changes are nice and the test works as expected. However the code is (in my opinion, at least) a little difficult to understand due to the lack of whitespace (see [PEP8](https://www.python.org/dev/peps/pep-0008/#other-recommendations) - Pycharm can do this automatically for you). Could this possibly be made a little easier on the eye?

File: scripts/Muon/fft_model.py (hunk 1 above)
Comment (AnthonyLim23) [reply scripts/Muon/fft_model.py]: this is for when a user cancels the algorithm. It still says that a user cancelled it but removes the stack trace

File: scripts/Muon/fft_model.py (hunk 2 above)
Comment (AnthonyLim23) [reply scripts/Muon/fft_model.py]: will be able to do that when I get back in Jan

**Comment by josephframsay** [comment]
Yup, everything looks good now!

**Review by josephframsay — APPROVED** [review]

**Pull request merged**